blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/http/HttpEnumIntegrationTest.java

```diff
@@ -50,12 +64,56 @@ public void testMetricIngestionWithEnum() throws Exception {
 
         // query for metric and assert results
         HttpResponse query_response = queryMetricIncludeEnum(tenant_id, metric_name);
-        Assert.assertEquals("Should get status 200 from query server for GET", 200, query_response.getStatusLine().getStatusCode());
+        assertEquals( "Should get status 200 from query server for GET", 200, query_response.getStatusLine().getStatusCode() );
 
         // assert response content
         String responseContent = EntityUtils.toString(query_response.getEntity(), "UTF-8");
-        Assert.assertEquals(String.format("[{\"metric\":\"%s\",\"enum_values\":[\"v1\",\"v2\",\"v3\"]}]", metric_name), responseContent);
+        assertEquals( String.format( "[{\"metric\":\"%s\",\"enum_values\":[\"v1\",\"v2\",\"v3\"]}]", metric_name ), responseContent );
         EntityUtils.consume(query_response.getEntity());
     }
 
+    @Test
+    public void testHttpEnumIngestionInvalidPastCollectionTime() throws IOException, URISyntaxException {
+
+        long timestamp = System.currentTimeMillis() - TIME_DIFF - Configuration.getInstance().getLongProperty( CoreConfig.BEFORE_CURRENT_COLLECTIONTIME_MS );
+
+        // ingest and rollup metrics with enum values and verify CF points and elastic search indexes
+        final String tenant_id = "333333";
+
+        String prefix = getPrefix();
+
+        final String metric_name = prefix + "enum_metric_test";
+        Set<Locator> locators = new HashSet<Locator>();
+        locators.add(Locator.createLocatorFromPathComponents(tenant_id, metric_name));
+
+        // post enum metric for ingestion and verify
+        HttpResponse response = postMetric(tenant_id, postAggregatedPath, "sample_enums_payload.json", timestamp, prefix );
+
+        String[] errors = getBodyArray( response );
+
+        assertEquals( 400, response.getStatusLine().getStatusCode() );
+        assertTrue( Pattern.matches( JSONMetricsContainerTest.PAST_COLLECTION_TIME_REGEX, errors[ 1 ] ) );
```

Comment:
Nit pick: maybe move this PAST_COLLECTION_TIME_REGEX to a Util class or base Test class?